src/streamlink/plugins/vk.py

```diff
@@ -25,6 +26,34 @@
 ))
 class VK(Plugin):
     API_URL = "https://vk.com/al_video.php"
+    HASH_COOKIE = "hash429"
+    SOLUTION_COOKIE = "solution429"
+
+    def get_saved_waf_cookie(self):
+        if not self.session.http.cookies.get(self.SOLUTION_COOKIE):
+            log.debug("Saved cookie not found.")
+            return
+        log.info("Using saved cookie.")
+        return True
+
+    def get_and_save_new_waf_cookie(self):
+        log.info("Getting new cookie.")
+        res = self.session.http.get("https://vk.com/")
+        cookie = self.session.http.cookies.get(self.HASH_COOKIE)
+        if not cookie:
+            log.error("Failed to get expected hash cookie.")
+            return
+        key = md5(cookie.encode("utf-8")).hexdigest()
+        res = self.session.http.get(f"{res.url}&key={key}")
```

Comment:
The `res` var is unused.

Setting the `key` parameter on the `params` keyword is also better than modifying the request URL.

Also, if you want to make mypy happy, then return False where you currently return None (the linting rule is currently disabled, hence why there's no failure).